test/test.js

```diff
@@ -0,0 +1,159 @@
+      assert.equal(body['style'], STYLE);
+      assert.equal(body['base_ccy'], BASE_CCY);
+      assert.equal(body['base_price'], BASE_PRICE);
+      assert.equal(body['remaining'], body['invoice_price']);
+      
+      var expiration = new Date(body['expiration']),
+          server_time = new Date(body['server_time'])
+      assert.isTrue(expiration > server_time)
+      
+      assert.lengthOf(body['id'], 26);
+      
+      var i = body['id'].length;
+      while (i--) {
+        assert.include('ABCDEFGHJKLMNPQRSTUVWXYZ23456789', body['id'][i]);
+      }
+      
+
+    });
+  });
+  
+  
+  it('should return authentication error', function(){
+    payload = {
+            base_ccy   : BASE_CCY,
+            base_price : BASE_PRICE,
+            callback   : CALLBACK,
+            style      : STYLE,
+            order      : ORDER
+    };
+    
+    yellow.createInvoice(DEFAULT_KEY, DEFAULT_SECRET, payload, function(error, response, body){
+      assert.notEqual(response.statusCode, 200)
+      
+    })
+  });
+
+  it('should return nonce error', function(){
+    payload = {
+            base_ccy   : BASE_CCY,
+            base_price : BASE_PRICE,
+            callback   : CALLBACK,
+            style      : STYLE,
+            order      : ORDER
+    };
+    
+    fakeTime = sinon.useFakeTimers(new Date(1992,12,29).getTime());
+    yellow.createInvoice(API_KEY, API_SECRET, payload, function(error, response, body){
+      assert.notEqual(response.statusCode, 200)
+      
+    });
+    fakeTime.restore();
+  });
+  
+  it('should return minimum price error', function(){
+    payload = {
+            base_ccy   : BASE_CCY,
+            base_price : '0.01',
+            callback   : CALLBACK,
+            style      : STYLE,
+            order      : ORDER
+    };
+    
+    yellow.createInvoice(API_KEY, API_SECRET, payload, function(error, response, body){
+      assert.notEqual(response.statusCode, 200)
+      
+    })
+  });
+  
+  it('should return base_ccy error', function(){
+    payload = {
+            base_ccy   : 'xxx',
+            base_price : BASE_PRICE,
+            callback   : CALLBACK,
+            style      : STYLE,
+            order      : ORDER
+    };
+    
+    yellow.createInvoice(API_KEY, API_SECRET, payload, function(error, response, body){
+      assert.notEqual(response.statusCode, 200)
+      
+    })
+  });
+  
+  it('should return callback error', function(){
+    payload = {
+            base_ccy   : BASE_CCY,
+            base_price : BASE_PRICE,
+            callback   : 'xxx',
+            style      : STYLE,
+            order      : ORDER
+    };
+    
+    yellow.createInvoice(API_KEY, API_SECRET, payload, function(error, response, body){
+      assert.notEqual(response.statusCode, 200)
+      
+    })
+  });
+  
+});
+
+
+
+describe('queryInvoice()', function(){
+  it('should query invoice', function(){
+    yellow.queryInvoice(API_KEY, API_SECRET, TEST_INVOICE_ID, function(error, response, body){
```

Comment:
Can you have this query the invoice that was created before? Otherwise we won't be able to run these tests on anything but the production server (i.e. it's onlyt he production server that will have invoice YBN4YC9FNMCPYMQZY3F8X55W9Y)